src/Engine/Renderer/RenderTechnique/ShaderProgram.hpp

```diff
@@ -63,9 +63,25 @@ class RA_ENGINE_API ShaderProgram final {
 
     void setUniform( const char* name, Texture* tex, int texUnit ) const;
 
+    //! use automatic texture unit computation
+    //! if you want to send a particular texture unit, use setUniform.
+    //! It binds tex on an "arbitrary" tex unit.
+    //! @warning, call a std::map::find (in O(log(active tex unit in the shader)))
+    void setUniformTexture( const char* texName, Texture* tex ) const;
+
     globjects::Program* getProgramObject() const;
 
   private:
+    struct TextureBinding {
+        int texUnit_;
+        int location_;
+        TextureBinding( int unit, int location ) : texUnit_( unit ), location_( location ) {}
+        TextureBinding() : texUnit_( -1 ), location_( -1 ) {}
+    };

```

Comment:
C.48: Prefer in-class initializers to member initializers in constructors for constant initializers
CPP core guidelines : https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers